### ISSUE TITLE
Use OpenSSL main cmake target in target link libraries for conan compatibility

### ISSUE THREAD

**Issue by egendron93** [edenhill/librdkafka#2614]
Description
===========
While working on an updated conan recipe that uses the experimental cmake in rdkafka, I came across the issue of the specified sub cmake targets in the OpenSSL cmake target not matching with the conan generated ones. I believe this issue would be resolved by adding the base cmake target of OpenSSL::OpenSSL instead of the specific sub targets of OpenSSL::SSL and OpenSSL::Crypto on [src/CMakeLists.txt Line 182](https://github.com/edenhill/librdkafka/blob/8681f884020e880a4c6cda3cfc672f0669e1f38e/src/CMakeLists.txt#L182). This is the same concept that is used when adding the zlib target when the zlib package is found.

I created this issue for v1.0.1 but i would like to see it in other versions too so the corresponding conan recipes can be created. For anyone curious and familiar with conan, I am using the cmake_find_package generator. The cmake_find_package generator generates the FindLib.cmake files for the recipes required dependencies and updates the CMAKE_MODULE_PATH so that the cmake find_package function can see the FindLib.cmake files generated by conan.

In summary simply changing line 182 in src/CmakeLists.txt from
```target_link_libraries(rdkafka PUBLIC OpenSSL::SSL OpenSSL::Crypto)```
to
```target_link_libraries(rdkafka PUBLIC OpenSSL::OpenSSL)```
would allow for a simple conan recipe to be created.

I would be happy to submit a PR

Thank you

Checklist
=========

 - [x] librdkafka version (release number or git tag): v1.0.1
 - [ ] Apache Kafka version: NA
 - [ ] librdkafka client configuration: NA
 - [ ] Operating system: NA
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka: NA
 - [ ] Provide broker log excerpts: NA
 - [ ] Critical issue

**Comment by egendron93** [comment]
I realized this solution will not work. The default cmake openssl find package does not provide a parent OpenSSL::OpenSSL target that contains the sub OpenSSL::SSL and OpenSSL::Crypto targets. The OpenSSL package in the stable conan repository does provide this parent target that encompasses both sub targets.
